Sources/ApolloCodegenLib/IR/IR+InclusionConditions.swift

```diff
@@ -54,6 +54,21 @@ extension IR {
       self.conditions = [condition]
     }
 
+    /// A failable initializer to convert from an array of `CompilationResult.InclusionCondition`.
+    init?(_ conditions: [CompilationResult.InclusionCondition]?) {
```

Comment:
@AnthonyMDev - I've added this since we determined that `ScopeCondition` should match the inclusion conditions as well as the parent type. Problem is that the named fragment passed into the scope condition is from `CompilationResult` and the two `InclusionCondition` types can't be compared. This is a simple conversion similar to how we treat conversions for the `IR.InclusionCondition.Result` type.

Reply:
I wonder if we could instead have a custom `==` operator that will compare `CompilationResult.InclusionConditions` to `IR.InclusionConditions`. That might be a better solution than an initializer that should never be used?

Reply:
The problem with including this initializer is that it doesn't have an accurate 1:1 way to map. For `@skip` with no variable, we would be omitting the entire fragment. The `preconditionFailure` here seems way too easy to accidentally make occur and cause a crash.

I think that a custom matching function that is `fileprivate` to the file that is using it is a better approach. It's not an actual `==`, so I would make it a function with a more descriptive name.